### PR TITLE
[SYCL][FPGA] Changing the attribute max_private_copies to private_copes

### PR DESCRIPTION
**Pull request by mohammadfawaz**
The IR coming out of this does not change. This allows the llvm-spirv
translator to not be affected. This should probably be fixed in the future
(in the SPIRV spec) for consistency.

### CODE REVIEW

**Comment by MrSidims** [comment]
LGTM max_ here is really looking redundant.

Please sign-off your commit.

File: clang/lib/CodeGen/CodeGenModule.cpp [hunk 1]
```diff
@@ -3975,9 +3975,9 @@ void CodeGenModule::generateIntelFPGAAnnotation(
     llvm::APSInt BWAInt = BWA->getValue()->EvaluateKnownConstInt(getContext());
     Out << '{' << BWA->getSpelling() << ':' << BWAInt << '}';
   }
-  if (const auto *MCA = D->getAttr<IntelFPGAMaxPrivateCopiesAttr>()) {
+  if (const auto *MCA = D->getAttr<IntelFPGAPrivateCopiesAttr>()) {
     llvm::APSInt MCAInt = MCA->getValue()->EvaluateKnownConstInt(getContext());
-    Out << '{' << MCA->getSpelling() << ':' << MCAInt << '}';
+    Out << "{max_private_copies:" << MCAInt << '}';
```
Comment (AGindinson) [inline clang/lib/CodeGen/CodeGenModule.cpp]: This seems to be doing the opposite of what's intended.

File: clang/lib/CodeGen/CodeGenModule.cpp [hunk 2]
```diff
@@ -3975,9 +3975,9 @@ void CodeGenModule::generateIntelFPGAAnnotation(
     llvm::APSInt BWAInt = BWA->getValue()->EvaluateKnownConstInt(getContext());
     Out << '{' << BWA->getSpelling() << ':' << BWAInt << '}';
   }
-  if (const auto *MCA = D->getAttr<IntelFPGAMaxPrivateCopiesAttr>()) {
+  if (const auto *MCA = D->getAttr<IntelFPGAPrivateCopiesAttr>()) {
     llvm::APSInt MCAInt = MCA->getValue()->EvaluateKnownConstInt(getContext());
-    Out << '{' << MCA->getSpelling() << ':' << MCAInt << '}';
```
Comment (MrSidims) [inline clang/lib/CodeGen/CodeGenModule.cpp]: Why this change is required?

File: clang/lib/CodeGen/CodeGenModule.cpp (hunk 1 above)
Comment (MrSidims) [inline clang/lib/CodeGen/CodeGenModule.cpp]: If this change is required, then:
```suggestion
    Out << "{private_copies:" << MCAInt << '}';
```

Comment (MrSidims) [reply clang/lib/CodeGen/CodeGenModule.cpp]: Ah, I got it. In that way we don't want to change LLVM IR output. It's looking hacky, isn't it? :)

Comment (mohammadfawaz) [reply clang/lib/CodeGen/CodeGenModule.cpp]: Exactly.. As you mentioned, the "max" here is redundant, and is in fact misleading to customers. That's why we're getting rid of it. However, I'm keeping the IR the same for now because otherwise, we would have to change the llvm-spirv translator and the SPIRV spec to do things the right way. I should open a JIRA to track that effort, but this does not affect the customer flow.

Comment (AGindinson) [reply clang/lib/CodeGen/CodeGenModule.cpp]: >> However, I'm keeping the IR the same for now because otherwise, we would have to change the llvm-spirv translator and the SPIRV spec to do things the right way.

I'd argue that no changes would have to be made to the SPIR-V specification, as it doesn't say anything with regard to the LLVM IR layout. Would it be better to accompany this patch with a correspondent adjustment to the translator? I beliebe the fix would be trivial.

Comment (MrSidims) [reply clang/lib/CodeGen/CodeGenModule.cpp]: @mohammadfawaz please address @AGindinson comment. I'm OK to go either way.

Comment (mohammadfawaz) [reply clang/lib/CodeGen/CodeGenModule.cpp]: @MrSidims @AGindinson I addressed the translator issue on the private repo. I will pull back here when I get approval there.

**Comment by mohammadfawaz** [comment]
Moved to this: https://github.com/otcshare/llvm/pull/819
